parser/rule_parser.go

```diff
@@ -92,6 +92,18 @@ func (r *RuleParser) LoadRule(filepath string) ([]engine.Rule, map[string]provid
 			return nil, nil, fmt.Errorf("a Rule must have a single condition")
 		}
 
+		// IF there is a not, then we assume a single condition at this level and store it to be used in the default case.
+		// There may be a better way of doing this.
+		notKeywordRaw, ok := whenMap["not"]
+		not := false
+		if ok {
+			delete(whenMap, "not")
+			not, ok = notKeywordRaw.(bool)
+			if !ok {
+				return nil, nil, fmt.Errorf("not must a boolean, not %v", notKeywordRaw)
+			}
+		}
```

Comment:
This is the part I keep re-reading not fully understanding what is happening.

Reply:
For example...why are we deleting things from the `whenMap`?

Reply:
Added a comment, let me know if that makes more sense now?